kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -288,6 +304,60 @@ private BigQuery getBigQuery() {
     return bigQuery.updateAndGet(bq -> bq != null ? bq : newBigQuery());
   }
 
+  private void setTimePartitioningForTimestamp(PartitionedTableId.Builder builder, TimePartitioning timePartitioning,
+                                               Long timestamp) {
+    if (timePartitioning != null) {
+      switch (timePartitioning.getType()) {
+        case HOUR:
+          builder.setHourPartition(timestamp);
+          break;
+        case MONTH:
+          builder.setMonthPartition(timestamp);
+          break;
+        case YEAR:
+          builder.setYearPartition(timestamp);
+          break;
+        default:
+          builder.setDayPartition(timestamp);
+      }
+    } else {
+      builder.setDayPartition(timestamp);
+    }
+  }
+
+  private void setTimePartitioning(PartitionedTableId.Builder builder, TimePartitioning timePartitioning) {
+    if (timePartitioning != null) {
```

Comment:
Based on the above suggestion about initializing the partitioning type to DAY by default, we can take out the null check here.

Reply:
Initialisation to DAY completed along with removal of the NULL checks